### PR TITLE
Fixed name validation in prompt mode

### PR DESCRIPTION
**Pull request by hzlmn**
Project name not validated if in created prompt mode

### CODE REVIEW

File: create_aio_app/main.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 from pathlib import Path
 
 import click
-from cookiecutter.exceptions import OutputDirExistsException
+from cookiecutter.exceptions import (FailedHookException,
```
Comment (Arfey) [inline create_aio_app/main.py]: I prefer to write like this
```
from cookiecutter.exceptions import (
    FailedHookException,
    OutputDirExistsException,
)
```

File: create_aio_app/template/hooks/pre_gen_project.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+from create_aio_app.utils.config import name_type
+
+
+def validate_project_name(project_name: str) -> None:
+    try:
+        name_type(project_name)
+    except argparse.ArgumentTypeError as exc:
+        click.echo(str(exc), err=True)
+        sys.exit(1)
```
Comment (Arfey) [inline create_aio_app/template/hooks/pre_gen_project.py]: why do u use that ` sys.exit(1)`?

Comment (hzlmn) [reply create_aio_app/template/hooks/pre_gen_project.py]: It is preffered way to force cookiecutter stop generation. https://cookiecutter.readthedocs.io/en/latest/advanced/hooks.html#exit-with-an-appropriate-status

**Comment by Arfey** [comment]
thx 😀

**Pull request merged**